src/fiber/channel.rs

```diff
@@ -2439,224 +2553,173 @@ impl TlcState {
-                    .find_map(|tlc| match tlc {
-                        TlcKind::AddTlc(info) if info.tlc_id == *id => Some(info),
-                        _ => None,
-                    })
-            }
-            TLCId::Received(_id) => {
-                self.remote_pending_tlcs
-                    .tlcs()
-                    .iter()
-                    .find_map(|tlc| match tlc {
-                        TlcKind::AddTlc(info) if info.tlc_id == *id => Some(info),
-                        _ => None,
-                    })
-            }
-        }
+    pub fn add_offered_tlc(&mut self, tlc: TlcInfo) {
+        self.offered_tlcs.add_tlc(tlc);
     }
 
-    pub fn get_mut(&mut self, id: &TLCId) -> Option<&mut AddTlcInfo> {
-        match id {
-            TLCId::Offered(_id) => {
-                self.local_pending_tlcs
-                    .tlcs
-                    .iter_mut()
-                    .find_map(|tlc| match tlc {
-                        TlcKind::AddTlc(info) if info.tlc_id == *id => Some(info),
-                        _ => None,
-                    })
-            }
-            TLCId::Received(_id) => {
-                self.remote_pending_tlcs
-                    .tlcs
-                    .iter_mut()
-                    .find_map(|tlc| match tlc {
-                        TlcKind::AddTlc(info) if info.tlc_id == *id => Some(info),
-                        _ => None,
-                    })
-            }
-        }
-    }
-
-    pub fn add_local_tlc(&mut self, tlc_info: TlcKind) {
-        self.local_pending_tlcs.push(tlc_info);
+    pub fn add_received_tlc(&mut self, tlc: TlcInfo) {
+        self.received_tlcs.add_tlc(tlc);
     }
 
-    pub fn add_remote_tlc(&mut self, tlc_info: TlcKind) {
-        self.remote_pending_tlcs.push(tlc_info);
-    }
-
-    pub fn next_local_tlc_id(&self) -> u64 {
-        self.local_pending_tlcs.next_tlc_id()
-    }
-
-    pub fn increment_local_tlc_id(&mut self) {
-        self.local_pending_tlcs.increment_next_tlc_id();
+    pub fn set_received_tlc_removed(
+        &mut self,
+        tlc_id: u64,
+        remove_at: CommitmentNumbers,
+        reason: RemoveTlcReason,
+    ) {
+        for tlc in self.received_tlcs.iter_mut() {
+            if Into::<u64>::into(tlc.tlc_id) == tlc_id {
+                tlc.removed_at = Some((remove_at, reason.clone()));
+                tlc.status = TlcStatus::Inbound(InboundTlctatus::LocalRemoved);
+            }
+        }
     }
 
-    pub fn next_remote_tlc_id(&self) -> u64 {
-        self.remote_pending_tlcs.next_tlc_id()
+    pub fn set_offered_tlc_removed(
+        &mut self,
+        tlc_id: u64,
+        remove_at: CommitmentNumbers,
+        reason: RemoveTlcReason,
+    ) {
+        for tlc in self.offered_tlcs.iter_mut() {
+            if Into::<u64>::into(tlc.tlc_id) == tlc_id {
+                tlc.removed_at = Some((remove_at, reason.clone()));
+                tlc.status = TlcStatus::Outbound(OutboundTlcStatus::RemoteRemoved);
```

Comment:
The status `RemoteRemoved` means remote has sent us a `RemoveTLC` and `CommitmentSigned` that confirms remote's intention to remove this TLC. The caller of the function `set_offered_tlc_removed` `handle_remove_tlc_peer_message` does not have any implication on whether there is a commitment transaction from the remote party to remove this TLC.

Reply:
The comment here may be wrong. It seems that `RemoteRemoved` only means remote has sent the `RemoveTlc` message (there is no implication on whether it is covered by a commitment transaction whatsoever). Only the state `RemoveWaitAck` implies that the remote sent a `CommitmentSigned` message **and** we have replied a `RevokeAndAck`. I think we have to document the meaning of each state.

Reply:
Ok, I will add more documents for each status